### PR TITLE
updates hapi-fhir service to use keycloak dns

### PR DESCRIPTION
**Pull request by ddjain**
- Updates docker-compose.yml
- Tested by locally building and running the application

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -24,17 +24,17 @@ services:
...
-      OAUTH_TOKEN_URL: https://auth-internal.elimuinformatics.com/auth/realms/product/protocol/openid-connect/token
-      OAUTH_MANAGE_URL: https://auth-internal.elimuinformatics.com/auth/realms/product/account
-      SMART_ISSUER: https://auth-internal.elimuinformatics.com/auth/realms/product
-      SMART_JWKS_URL: https://auth-internal.elimuinformatics.com/auth/realms/product/protocol/openid-connect/certs
-      SMART_AUTHORIZATION_URL: https://auth-internal.elimuinformatics.com/auth/realms/product/protocol/openid-connect/auth
-      SMART_TOKEN_URL: https://auth-internal.elimuinformatics.com/auth/realms/product/protocol/openid-connect/token
-      SMART_INTROSPECTION_URL: https://auth-internal.elimuinformatics.com/auth/realms/product/protocol/openid-connect/token/introspect
-      SMART_REVOCATION_URL: https://auth-internal.elimuinformatics.com/auth/realms/product/protocol/openid-connect/revoke
-      SMART_MANAGE_URL: https://auth-internal.elimuinformatics.com/auth/realms/product/account
+      OAUTH_JWKS_URL: https://auth-stage.elimuinformatics.com/auth/realms/product/protocol/openid-connect/certs
```
Comment (hankwallace) [inline docker-compose.yml]: I don't think we should commit these changes. Instead, I think this should use a local keycloak. Running keycloak on a local system should not use our stage environment's keycloak server

Comment (hankwallace) [reply docker-compose.yml]: I propose closing this pull request or updating the docker compose to use a local keycloak container.

Comment (ddjain) [reply docker-compose.yml]: Okay, I will update to use local keycloak container

File: docker-compose.yml [hunk 2]
```diff
@@ -24,17 +24,17 @@ services:
...
+      OAUTH_AUTHORIZE_URL: http://keycloak-url/auth/realms/product/protocol/openid-connect/auth
+      OAUTH_TOKEN_URL: http://keycloak-url/auth/realms/product/protocol/openid-connect/token
+      OAUTH_MANAGE_URL: http://keycloak-url/auth/realms/product/account
+      SMART_ISSUER: http://keycloak-url/auth/realms/product
+      SMART_JWKS_URL: http://keycloak-url/auth/realms/product/protocol/openid-connect/certs
+      SMART_AUTHORIZATION_URL: http://keycloak-url/auth/realms/product/protocol/openid-connect/auth
+      SMART_TOKEN_URL: http://keycloak-url/auth/realms/product/protocol/openid-connect/token
+      SMART_INTROSPECTION_URL: http://keycloak-url/auth/realms/product/protocol/openid-connect/token/introspect
+      SMART_REVOCATION_URL: http://keycloak-url/auth/realms/product/protocol/openid-connect/revoke
+      SMART_MANAGE_URL: http://keycloak-url/auth/realms/product/account
```
Comment (dimon37) [inline docker-compose.yml]: If you are referencing keycloak-url, shouldn't you also add the keycloak-url as the service?

**Review by dimon37 — CHANGES_REQUESTED** [review]

File: docker-compose.yml (hunk 2 above)
Comment (ddjain) [reply docker-compose.yml]: I have updated docker-compose file

File: docker-compose.yml [hunk 3]
```diff
@@ -51,6 +51,34 @@ services:
...
+      KC_HEALTH_ENABLED: 'true'
+      KC_HOSTNAME: 'keycloak:8085'
+      KC_HOSTNAME_STRICT: 'false'
+      KC_HTTP_RELATIVE_PATH: '/auth'
+      KC_HTTP_ENABLED: 'true'
+      KC_LOG_LEVEL: 'INFO'
+      KC_METRICS_ENABLED: 'true'
+      KC_PROXY: 'edge'
+      PROXY_ADDRESS_FORWARDING: 'true'
+      HEADER_NAME: 'patient'
```
Comment (hankwallace) [inline docker-compose.yml]: This is no longer needed or valid.

File: docker-compose.yml [hunk 4]
```diff
@@ -51,6 +51,34 @@ services:
...
+      TENANT_HEADER_NAME: 'tenant'
```
Comment (hankwallace) [inline docker-compose.yml]: This is no longer needed or valid.

File: docker-compose.yml [hunk 5]
```diff
@@ -51,6 +51,34 @@ services:
...
+    restart: on-failure
+    ports:
+      - '8085:8085'
+    command:
+      - "start-dev"
+    environment:
+      KC_ADMIN: 'admin'
+      KC_ADMIN_PASSWORD: 'admin'
+      KC_DB: 'postgres'
+      KC_DB_URL_HOST: 'hapi-fhir-postgres'
```
Comment (hankwallace) [inline docker-compose.yml]: I don't think that the two different containers can use the same postgres.

File: docker-compose.yml (hunk 3 above)
Comment (ddjain) [reply docker-compose.yml]: Okay, I will remove it then

File: docker-compose.yml (hunk 5 above)
Comment (ddjain) [reply docker-compose.yml]: I will add another postgress container for keycloak

File: docker-compose.yml [hunk 6]
```diff
@@ -89,6 +86,20 @@ services:
       POSTGRES_PASSWORD: "admin"
     volumes:
       - hapi-fhir-postgres:/var/lib/postgresql/data
+  keycloak-postgres:
+    image: postgres:15
+    container_name: keycloak-postgres
+    restart: always
+    ports:
+      - "5432:5432"
```
Comment (hankwallace) [inline docker-compose.yml]: Aren't both postgres containers using this port? Did you test this on your local system?

Comment (ddjain) [reply docker-compose.yml]: I was initially testing with MySQL and PostgreSQL while running docker-compose up. During this process, I encountered some Keycloak errors, so I added the port mappings to test certain Keycloak changes separately. Unfortunately, those modifications were inadvertently committed.

**Review by hankwallace — APPROVED** [review]

**Review by dimon37 — APPROVED** [review]
Looks reasonable

**Pull request merged**